src/actions.sol

```diff
@@ -68,98 +118,103 @@ contract Actions {
         if (amount > debt) {
             amount = debt;
         }
 
-        _repay(shelf, erc20, loan, amount);
+        _repay(erc20, loan, amount);
     }
 
-    function repayFullDebt(address shelf, address pile, address erc20, uint256 loan) public {
-        _repay(shelf, erc20, loan, PileLike(pile).debt(loan));
+    function repayFullDebt(address erc20, uint256 loan) public onlyDelegateCall {
+        _repay(erc20, loan, PileLike(pile).debt(loan));
     }
 
-    function _repay(address shelf, address erc20, uint256 loan, uint256 amount) internal {
+    function _repay(address erc20, uint256 loan, uint256 amount) internal {
         // transfer money from borrower to proxy
         ERC20Like(erc20).transferFrom(msg.sender, address(this), amount);
         ERC20Like(erc20).approve(address(shelf), amount);
         ShelfLike(shelf).repay(loan, amount);
         emit Repay(shelf, erc20, loan, amount);
     }
 
-    function unlock(address shelf, address registry, uint256 token, uint256 loan) public {
+    function unlock(address registry, uint256 token, uint256 loan) public onlyDelegateCall {
         ShelfLike(shelf).unlock(loan);
         NFTLike(registry).transferFrom(address(this), msg.sender, token);
         emit Unlock(shelf, registry, token, loan);
     }
 
-    function close(address shelf, uint256 loan) public {
+    function close(uint256 loan) public onlyDelegateCall {
         ShelfLike(shelf).close(loan);
         emit Close(shelf, loan);
     }
 
     // --- Borrower Wrappers ---
-    function transferIssue(address shelf, address registry, uint256 token) public returns (uint256 loan) {
+    function transferIssue(address registry, uint256 token) public onlyDelegateCall returns (uint256 loan) {
         transfer(registry, token);
-        return issue(shelf, registry, token);
-    }
-
-    function lockBorrowWithdraw(address shelf, uint256 loan, uint256 amount, address usr) public {
-        lock(shelf, loan);
-        borrowWithdraw(shelf, loan, amount, usr);
+        return issue(registry, token);
     }
 
-    function transferIssueLockBorrowWithdraw(
-        address shelf,
-        address registry,
-        uint256 token,
-        uint256 amount,
-        address usr
-    ) public {
-        uint256 loan = transferIssue(shelf, registry, token);
-        lockBorrowWithdraw(shelf, loan, amount, usr);
+    function lockBorrowWithdraw(uint256 loan, uint256 amount) public onlyDelegateCall {
+        lock(loan);
+        borrowWithdraw(loan, amount);
     }
 
-    function repayUnlockClose(address shelf, address pile, address registry, uint256 token, address erc20, uint256 loan)
+    function mintIssuePriceLock(address registry, uint256 price, uint256 riskGroup)
         public
+        onlyDelegateCall
+        returns (uint256 loan, uint256 tokenId)
     {
-        repayFullDebt(shelf, pile, erc20, loan);
-        unlock(shelf, registry, token, loan);
-        close(shelf, loan);
+        tokenId = mintAsset(registry);
```

Comment:
if we don't give direct auth access to the nftAsset contract, the registry and minter contract should be different